rotkehlchen/chain/evm/decoding/socket_bridge/decoder.py

```diff
@@ -0,0 +1,115 @@
+
+
+class SocketBridgeDecoder(DecoderInterface):
+    """The gateway contract is deployed in all the chains with the same address"""
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.eth = A_ETH.resolve_to_crypto_asset()
+
+    def _decode_bridged_asset(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] != BRIDGE_TOPIC:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[0:32])
+        token_address = hex_or_bytes_to_address(context.tx_log.data[32:64])
+
+        if token_address == ETH_SPECIAL_ADDRESS:
+            bridged_asset: 'CryptoAsset' = self.eth
+        else:
+            bridged_asset = get_or_create_evm_token(
+                userdb=self.evm_inquirer.database,
+                evm_address=token_address,
+                chain_id=self.evm_inquirer.chain_id,
+                evm_inquirer=self.evm_inquirer,
+            )
+        amount = asset_normalized_value(amount=amount_raw, asset=bridged_asset)
+        sender = hex_or_bytes_to_address(context.tx_log.data[128:160])
+        receiver = hex_or_bytes_to_address(context.tx_log.data[160:192])
+        to_chain_id_raw = hex_or_bytes_to_int(context.tx_log.data[64:96])
+
+        try:
+            to_chain_id = ChainID.deserialize_from_db(to_chain_id_raw)
+            _, target_chain = to_chain_id.name_and_label()
+        except DeserializationError:
+            target_chain = str(to_chain_id_raw)
+            log.error(f'Unknown chain in socket bridge {to_chain_id_raw}')
+
+        for event in context.decoded_events:
+            if (
+                event.location_label == sender and
+                event.asset == bridged_asset and
+                event.balance.amount == amount and
+                event.event_type == HistoryEventType.SPEND
+            ):
+                if self.base.is_tracked(receiver):  # if receiver is not tracked we are spending it  # noqa: E501
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.BRIDGE
+
+                event.counterparty = CPT_SOCKET
+                _, from_chain = self.evm_inquirer.chain_id.name_and_label()
+                event.notes = (
+                    f'Bridge {amount} {bridged_asset.symbol} from {from_chain} '
+                    f'to {receiver} at {target_chain} using Socket'
+                )
+
+        return DEFAULT_DECODING_OUTPUT
+
+    # -- DecoderInterface methods
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            GATEWAY_ADDRESS: (self._decode_bridged_asset,),
```

Comment:
This seems to only handle bridging to another chain. Can you also handle the bridge receiving (withdrawal) case?

Reply:
As discussed in the standup it delegates the bridging to other sources and those handle the withdrawal